### PR TITLE
Philip Gheno - Task Manager

### PR DESCRIPTION
**Pull request by minijag**
I may have not included a ProjectList class.  Otherwise you may notice how legendary my code is

### CODE REVIEW

File: lib/task-manager/pm.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+require_relative '../task-manager.rb'
+
+
+class TM::PM
+  @exit = false
+
+
+  def main_menu
+
+    while !@exit do
```
Comment (parthshahva) [inline lib/task-manager/pm.rb:11]: consider making this menu a little easier to read in the terminal

File: lib/task-manager/pm.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  end
+
+  def thatUniqueProject(projectID)
+    TM::Project.uniqueProjectArray.each { |index|
+      if index.uniqueId == projectID.to_i
+        @projectInstance = index
+        return @projectInstance
+      end
+    }
+    puts "Cant find project instance with that ID"
```
Comment (parthshahva) [inline lib/task-manager/pm.rb:42]: I think this line will run regardless of whether a project is found or not, but I could be wrong

File: lib/task-manager/pm.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      puts "So many projects"
+
+    elsif input1 == 'history'
+      thatUniqueProject(input2)
+      @projectInstance.addTask
+      puts @projectInstance.completedTaskArray
+
+    elsif input1 == 'addtask'
+      thatUniqueProject(input2)
+      puts "What should your projects Title be?"
```
Comment (parthshahva) [inline lib/task-manager/pm.rb:74]: nice

File: lib/task-manager/pm.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      puts "      mark PID TID - Mark a task in its project with its TID(TaskID) as complete"
+      puts "      quit -- get out of the progam"
+      puts "    What can I do for you today?"
+      userinput = gets.chomp.downcase
+      puts "Command Accepted '#{userinput}'"
+      parseInput(userinput)
+    end
+  end
+
+  def parseInput(userInput)
```
Comment (parthshahva) [inline lib/task-manager/pm.rb:29]: the methods that follow should be tested?

File: lib/task-manager.rb [hunk 5]
```diff
@@ -6,3 +6,5 @@ module TM
 # Require all of our project files
```
Comment (parthshahva) [inline lib/task-manager.rb:6]: GENERAL COMMENTS
- I like how you take user input and make it easier for users to get to the commands to create and access data
- interesting how you store data without a middleman, maybe a middleman would make it easier to handle the data?
- keep methods away from the client side or place them within a middleman class possibly?
- write more tests for the methods that receive user input
- code was very clean and easy to read, Awesome work man!

File: lib/task-manager/project.rb [hunk 6]
```diff
@@ -1,3 +1,61 @@
...
+
+
+  def addTask(name='noname', description='no description', priority=1)
+    @uniqueTaskCounter += 1
+    newtask = TM::Task.new(@uniqueTaskCounter, name, description, priority)
+    @taskArray << newtask
+  end
+
+  def changeStatus(thattaskID, newstatus)
+    taskArray.each do |index|
```
Comment (parthshahva) [inline lib/task-manager/project.rb:30]: use the select method here possibly?

File: lib/task-manager/task.rb [hunk 7]
```diff
@@ -1,3 +1,20 @@
 
 class TM::Task
+
+  attr_reader :taskName, :description, :uniqueTaskId, :creationTime
+  attr_accessor :status, :priority
+  @@dattime = Time.now
+
+  def initialize(uniqueTaskCounter, taskName='noname', description='no description', priority=1, status='incomplete')
```
Comment (parthshahva) [inline lib/task-manager/task.rb:8]: did not leave a id relating to project

File: lib/task-manager/pm.rb [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    parsedInput = userInput.split( " " )
+
+    runCommand(parsedInput[0], parsedInput[1], parsedInput[2])
+  end
+
+  def thatUniqueProject(projectID)
+    TM::Project.uniqueProjectArray.each { |index|
```
Comment (gilbert) [inline lib/task-manager/pm.rb:36]:
``` ruby
unique_project = TM::Project.uniqueProjectArray.find do |project|
  project.uniqueId == projectID.to_i
end

puts "Can't find project instance with that ID" if unique_project.nil?
```

File: lib/task-manager/project.rb [hunk 9]
```diff
@@ -1,3 +1,61 @@
...
+      puts x.creationTime
+    }
+  end
+
+  def sortTasksbyPriority(whichArray)
+    whichArray.sort! { |x, y| (x.priority <=> y.priority) == 0 ? x.creationTime <=> y.creationTime :  y.priority <=> x.priority }
+  end
+
+  def grabCompletedTasks
+    @taskArray.each { |x|
```
Comment (gilbert) [inline lib/task-manager/project.rb:50]: Try using `@taskArray.find`, as well as using the `find` method for any time you're looking for a single thing in an array.

**Comment by gilbert** [comment]
Great job, Philip

**Pull request closed without merging**